### PR TITLE
add docs on migrating block documents

### PR DESCRIPTION
**Pull request by zzstoatzz**
### Example

Shows how to use the new block document / schema migration built into `Block.save` / `Block.load`
### Checklist

- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

File: docs/concepts/blocks.md [hunk 1]
```diff
@@ -286,6 +286,35 @@ my_s3_bucket.save("my_s3_bucket")
 
 In the above example, the values for `AWSCredentials` are saved with `my_s3_bucket` and will not be usable with any other blocks.
 
+### Handling updates to custom `Block` types
+Let's say that you now want to add a field `bucket_folder` to your custom `S3Bucket` block that represents the default path to read and write objects from (this fields exists on [our implementation](https://github.com/PrefectHQ/prefect-aws/blob/main/prefect_aws/s3.py#L292)).
```
Comment (tpdorsey) [inline docs/concepts/blocks.md:290]:
```suggestion
Let's say that you now want to add a `bucket_folder` field to your custom `S3Bucket` block that represents the default path to read and write objects from (this field exists on [our implementation](https://github.com/PrefectHQ/prefect-aws/blob/main/prefect_aws/s3.py#L292)).
```

File: docs/concepts/blocks.md [hunk 2]
```diff
@@ -286,6 +286,35 @@ my_s3_bucket.save("my_s3_bucket")
...
+
+
+```python hl_lines="4"
+class S3Bucket(Block):
+    bucket_name: str
+    credentials: AWSCredentials
+    bucket_folder: str = None
+    ...
+```
+...and [register the updated block type with your Prefect Orion server](#registering-blocks-for-use-in-the-prefect-ui) (whether cloud or self-hosted).
```
Comment (tpdorsey) [inline docs/concepts/blocks.md:302]:
```suggestion

Then [register the updated block type](#registering-blocks-for-use-in-the-prefect-ui) with either Prefect Cloud or your self-hosted Prefect Orion server.
```

**Review by tpdorsey — APPROVED** [review]
Looks good. Some minor suggested edits.

**Comment by madkinsz** [comment]
@zzstoatzz can you get this ready for release today?

**Review by ahuang11 — APPROVED** [review]

**Review by desertaxle — APPROVED** [review]
LGTM!

**Pull request merged**